### PR TITLE
More targeted debug and error messages

### PR DESCRIPTION
**Pull request by ejoerns**
This aims to improve debug and error message reporting in cases where we can give more useful information but do not do yet or where information may be misleading.
It also removes redundant infos and fixes usage of correct printing function.

### CODE REVIEW

**Comment by ejoerns** [comment]
@jluebbe I assumed this is basically described by commit message of d2c7725.

glib says:

> g_print() should not be used from within libraries for debugging messages, since it may be redirected by applications to special purpose message windows or even files. Instead, libraries should use g_log(), g_log_structured(), or the convenience macros g_message(), g_warning() and g_error().

File: src/install.c [hunk 1]
```diff
@@ -917,7 +917,7 @@ static gboolean launch_and_wait_default_handler(RaucInstallArgs *args, gchar* bu
 		else
 			g_message("Updating %s with %s", dest_slot->device, mfimage->filename);
 
-		r_context_begin_step("copy_image", "Copying image", 0);
+		r_context_begin_step("copy_image", g_strdup_printf("Copying image to %s", dest_slot->name), 0);
```
[the remark this answers is not in the document]
Comment (ejoerns) [reply src/install.c]: It's to be consistent with https://github.com/rauc/rauc/blob/master/src/install.c#L876 :)

[the remark this answers is not in the document]
Comment (ejoerns) [reply src/install.c]: Hm.. I fear this must be fixed at another level.

At the moment the progress function `r_context_being_step()` creates a new progress element based on the input strings and nesting depth and adds it to the list of progress steps.

But the strings the steps use are assigned with their original pointers. Thus heap-allocated strings have to remain valid until the entire progress is done.

To reduce code complexity, maybe it is worth adding a printf-style context function? And additionally use allocated strings inside the context handling.

[the remark this answers is not in the document]
Comment (jluebbe) [reply src/install.c]: Yes. Having it similar to g_message would make it easier to use.

[the remark this answers is not in the document]
Comment (ejoerns) [reply src/install.c]: I've created #345 to fix this.

**Comment by ejoerns** [comment]
Note that I've pushed an update to satisfy tests that complained about not seeing expected deprecation warning message. This was caused by switching one log message from `g_print()` to `g_message()` making it appear and being handled by log message check before expected  `g_warning()` occurs.
Adding the `g_message()` to the stack of expected log messages fixed it.

**Comment by ejoerns** [comment]
Rebased onto current master and made use of `r_context_begin_step_formatted()`

**Pull request merged**